plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityOpacity.java

```diff
@@ -0,0 +1,53 @@
+package com.denizenscript.denizen.objects.properties.entity;
+
+import com.denizenscript.denizen.objects.EntityTag;
+import com.denizenscript.denizencore.objects.Mechanism;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import org.bukkit.entity.TextDisplay;
+
+public class EntityOpacity extends EntityProperty<ElementTag> {
+
+    // <--[property]
+    // @object EntityTag
+    // @name opacity
+    // @input ElementTag(Number)
+    // @description
+    // A text display entity's text opacity, from 0 to 255.
+    // Can be interpolated.
+    // -->
+
+    public static boolean describes(EntityTag entity) {
+        return entity.getBukkitEntity() instanceof TextDisplay;
+    }
+
+    @Override
+    public ElementTag getPropertyValue() {
+        return new ElementTag(Byte.toUnsignedInt(as(TextDisplay.class).getTextOpacity()));
+    }
+
+    @Override
+    public boolean isDefaultValue(ElementTag value) {
+        return value.asInt() == 255;
+    }
+
+    @Override
+    public void setPropertyValue(ElementTag value, Mechanism mechanism) {
+        if (mechanism.requireInteger()) {
+            int opacity = value.asInt();
+            if (opacity < 0 || opacity > 255) {
+                mechanism.echoError("Invalid opacity specified, must be between 0 and 255.");
+                return;
+            }
+            as(TextDisplay.class).setTextOpacity((byte) opacity);

```

Comment:
I notice you convert from signed to unsigned in `get`, but not in `set` - is this intentional? Does this actually behave as expected?

Reply:
I'm pretty sure this is right? Correct me if I'm wrong, but according to my testing (and google) this should convert it properly (and it does look right in-game, other then the 0-3 edge case):
```
[19:34:34 INFO]: +> [EntityOpacity] Setting opacity: 255
[19:34:34 INFO]: +> [EntityOpacity] Raw opacity: -1

[19:34:40 INFO]: +> [EntityOpacity] Setting opacity: 128
[19:34:40 INFO]: +> [EntityOpacity] Raw opacity: -128

[19:34:43 INFO]: +> [EntityOpacity] Setting opacity: 127
[19:34:43 INFO]: +> [EntityOpacity] Raw opacity: 127

[19:34:58 INFO]: +> [EntityOpacity] Setting opacity: 234
[19:34:58 INFO]: +> [EntityOpacity] Raw opacity: -22
```